### PR TITLE
Set of consumer-prefix-group from relation data

### PR DESCRIPTION
**Pull request by welpaolo**
Addresses [DPE-1342](https://warthogs.atlassian.net/browse/DPE-1342).

[DPE-1342]: https://warthogs.atlassian.net/browse/DPE-1342?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/integration/test_provider.py [hunk 1]
```diff
@@ -244,6 +244,9 @@ async def test_connection_updated_on_tls_enabled(ops_test: OpsTest, app_charm: P
         model_full_name=ops_test.model_full_name,
         endpoint="kafka-client-consumer",
     )
+
     assert provider_data["tls"] == "enabled"
     assert "9093" in provider_data["endpoints"]
     assert "2182" in provider_data["zookeeper-uris"]
+    assert "test-prefix" in provider_data["consumer-group-prefix"]
+    assert "test-topic" in provider_data["topic"]
```
Comment (marcoppenheimer) [inline tests/integration/test_provider.py:252]: I think you could add a check in L69-L74 checking for the expected CONSUMER_GROUP_PREFIX in the ACLs returned, just to confirm Kafka actually set the right ones.

**Review by marcoppenheimer — CHANGES_REQUESTED** [review]

File: lib/charms/data_platform_libs/v0/data_interfaces.py [hunk 2]
```diff
@@ -304,7 +303,9 @@ def _on_topic_requested(self, event: TopicRequestedEvent):
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 5
+LIBPATCH = 8
+
+PYDEPS = ["ops>=2.0.0"]
```
[the remark this answers is not in the document]
Comment (welpaolo) [reply lib/charms/data_platform_libs/v0/data_interfaces.py:308]: This specifies dependencies in charm libraries: https://discourse.charmhub.io/t/python-dependencies-in-charm-libraries/8599

File: lib/charms/data_platform_libs/v0/data_interfaces.py [hunk 3]
```diff
@@ -544,7 +539,19 @@ def _diff(self, event: RelationChangedEvent) -> Diff:
...
+            if self._is_relation_active(relation)
+        ]
+
+    @staticmethod
+    def _is_relation_active(relation: Relation):
+        try:
+            _ = repr(relation.data)
+            return True
+        except RuntimeError:
+            return False
```
[the remark this answers is not in the document]
Comment (welpaolo) [reply lib/charms/data_platform_libs/v0/data_interfaces.py:554]: The problem here is was that relation data could not be read `on-relation-broken`. This code was not added for Kafka and it was reviewed in another PR on the data-platform-libs repo.

File: lib/charms/data_platform_libs/v0/data_interfaces.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jnsgruk) [reply lib/charms/data_platform_libs/v0/data_interfaces.py:308]: Oh nice, I hadn't thought of using this to pin to a minimum ops version :+1:

**Review by deusebio — APPROVED** [review]
All good! 

@marcoppenheimer  In this PR only "src/provider.py" should be reviewed. The rest are changes of a library that are out-of-scope of this PR, but they are here only because of the update

File: tests/integration/test_provider.py (hunk 1 above)
Comment (welpaolo) [reply tests/integration/test_provider.py:252]: Done, thanks for the suggestion!

File: lib/charms/data_platform_libs/v0/data_interfaces.py [hunk 4]
```diff
@@ -1059,26 +1118,34 @@ class KafkaRequires(DataRequires):
...
-        )
+        relation_data = {}
+        if self.topic:
+            relation_data["topic"] = self.topic
+        if self.extra_user_roles:
+            relation_data["extra-user-roles"] = self.extra_user_roles
+        if self.consumer_group_prefix:
+            relation_data["consumer-group-prefix"] = self.consumer_group_prefix
+
+        self._update_relation_data(event.relation.id, relation_data)
```
[the remark this answers is not in the document]
Comment (welpaolo) [reply lib/charms/data_platform_libs/v0/data_interfaces.py:1148]: I like it, I will port this change in the PR opened in the data-platform-libs

[the remark this answers is not in the document]
Comment (welpaolo) [reply lib/charms/data_platform_libs/v0/data_interfaces.py:1148]: Unfortunately, this is not passing the unit tests. The empty string is considered `None`  by the testing framework

**Review by marcoppenheimer — APPROVED** [review]
LGTM :)

**Pull request merged**